R/govTable.R

```diff
@@ -31,91 +31,66 @@
 #'     shinyGovstyle::banner(
 #'       inputId = "banner", type = "beta", 'This is a new service'),
 #'     shinyGovstyle::gov_layout(size = "two-thirds",
-#'     shinyGovstyle::govTable(
-#'       "tab1", example_data, "Test", "l", num_col = c(2,3),
-#'       width_overwrite = c("one-half", "one-quarter", "one-quarter"))
+#'       govTable(
+#'         "tab1", example_data, "Test Table", "l",
+#'         right_col = c("Colours", "Bikes", "Cars", "Vans", "Buses"),
+#'         col_widths = list(Months = "one-third"),
+#'         page_size = 5
+#'         )
 #'     ),
 #'
 #'     shinyGovstyle::footer(full = TRUE)
 #'   )
 #'
 #'   server <- function(input, output, session) {}
 #'
-#'   shinyApp(ui = ui, server = server)
+#'   shinyApp(ui, server)
 #' }
 
 govTable <- function(inputId, df, caption, caption_size = "l",
-                     num_col = NULL, width_overwrite = NULL){
+                      right_col = NULL,
+                      col_widths = list(),
+                      page_size = 10) {
 
-  #Create row by row the main bulk of table to insert later
-  main_row_store <- NULL
-  for(i in 1:nrow(df)) {
-    temp_row_store <- createRows(df[i,], num_col)
-    main_row_store <- shiny::tagList(temp_row_store, main_row_store)
-  }
+  # Generate column definitions
+  col_defs <- setNames(lapply(seq_along(names(df)), function(index) {
```

Comment:
my devtools::check() throws up an issue here but I think using `stats::setNames(...` instead works to remove it if you don't mind adding in!